### PR TITLE
Add ScrubCalculatedFields to Resource interface

### PR DESCRIPTION
**Pull request by calufornia**
## Description

Adds the ability to scrub calculated fields from resources, the inverse operation to `Calculate()`. We'll call this method before saving resources to the dev config file.

## Test plan
Test with CLI changes that call `ScrubCalculatedFields()`

### CODE REVIEW

**Comment by calufornia** [comment]
Side note: I don't really like the empty `ScrubCalculatedFields` methods, and we should probably just have a default implementation on the `BaseResource` (same with `Calculate`).

File: pkg/resources/resource.go [hunk 1]
```diff
@@ -99,16 +99,15 @@ type Resource interface {
...
 	// Validate returns an error if the resource is invalid.
 	Validate() error
 	// Kind returns the ResourceKind associated with this resource.
 	// Deprecated: Use GetKind() instead
 	Kind() ResourceKind // TODO: Remove
 	// String returns a string representation of this resource.
 	String() string
-	// ID returns the resource's ID.
-	// Deprecated: Use GetID() instead
-	ID() string // TODO: Remove
```
Comment (calufornia) [inline pkg/resources/resource.go:111]: This was unused

File: pkg/resources/kinds/rest.go [hunk 2]
```diff
@@ -220,3 +227,6 @@ func (a *RESTAuthBasic) calculate() error {
 	}
 	return nil
 }
+
+// Do nothing — we can't determine if the Authorization header is calculated or defined by the user.
+func (a *RESTAuthBasic) scrubCalculatedFields() {}
```
Comment (francescaleung) [inline pkg/resources/kinds/rest.go:232]: headers in auth are always calculated--this should scrub the auth headers.

If folks want custom Authorization headers they shouldn't be using AuthBasic, just throw it in the regular REST headers.

File: pkg/resources/resource.go (hunk 1 above)
Comment (francescaleung) [reply pkg/resources/resource.go:111]: :tada:

**Review by francescaleung — APPROVED** [review]

**Pull request merged**